corehq/motech/repeaters/models.py

```diff
@@ -348,21 +352,26 @@ def _repeater_type(cls):
 
     @property
     def repeat_records_ready(self):
-        return self.repeat_records.filter(state__in=(State.Pending, State.Fail))
+        return (
+            self.repeat_records
+            .filter(state__in=(State.Pending, State.Fail))
+            .order_by('registered_at')
```

Comment:
I'm getting confused on `next_check` and where that is being set/checked, but is it possible that we would end up in a scenario where a repeater processes the same record multiple consecutive times because it's `registered_at` is the oldest, but it continues to fail? This would obviously have an upper limit of attempts (4 or 6), but I think would still not be ideal.

Reply:
If one payload fails repeatedly, but another succeeds, that sounds like a problem with the payload and not the remote endpoint. Trying to send it again isn't going to help. I think when a remote API responds with a 4XX error, we should immediately cancel the repeat record.

`RepeatRecord.next_check` is no longer relevant. It only exists because we had no way of figuring out what was happening with the remote endpoint. But now that we are iterating Repeaters instead of RepeatRecords, we can back off based on the status of the endpoint, and not the failure of the payload. Repeater back-off happens in the `update_repeater()` task.

Reply:
> I think when a remote API responds with a 4XX error, we should immediately cancel the repeat record.

What about a 429 (Too Many Requests) response with a `retry-after` header, where the remote API is asking us to back off? That seems like exactly the type of situation where we'd want to retry later.

Reply:
https://github.com/dimagi/commcare-hq/pull/34946/commits/5370a17967b0abd2d6fcd376511b86881ca8d87d